crates/gossip/src/summary.rs

```diff
@@ -0,0 +1,194 @@
+use crate::gossip::K2Gossip;
+use kitsune2_api::{DhtArc, K2Error, K2Result, LocalAgent, Timestamp, Url};
+use kitsune2_dht::{ArcSet, DhtSnapshot};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+/// DHT segment state.
+///
+/// See [`DhtSnapshot::Minimal`]
+#[derive(Debug, Serialize, Deserialize)]
+pub struct DhtSegmentState {
+    /// The top hash of the DHT ring segment.
+    pub disc_top_hash: bytes::Bytes,
+    /// The boundary timestamp of the DHT ring segment.
+    pub disc_boundary: Timestamp,
+    /// The top hashes of each DHT ring segment.
+    pub ring_top_hashes: Vec<bytes::Bytes>,
+}
+
+/// Peer metadata dump.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PeerMeta {
+    /// The timestamp of the last gossip round.
+    pub last_gossip_timestamp: Option<Timestamp>,
+    /// The bookmark of the last op bookmark received.
+    pub new_ops_bookmark: Option<Timestamp>,
+    /// The number of behavior errors observed.
+    pub peer_behavior_errors: Option<u32>,
+    /// The number of local errors.
+    pub local_errors: Option<u32>,
+    /// The number of busy peer errors.
+    pub peer_busy: Option<u32>,
+    /// The number of terminated rounds.
+    ///
+    /// Note that termination is not necessarily an error.
+    pub peer_terminated: Option<u32>,
+    /// The number of completed rounds.
+    pub completed_rounds: Option<u32>,
+    /// The number of peer timeouts.
+    pub peer_timeouts: Option<u32>,
+}
+
+/// Gossip round state summary.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct GossipRoundStateSummary {
+    /// The URL of the peer with which the round is initiated.
+    pub session_with_peer: Url,
+}
+
+/// Gossip state summary.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct GossipSummary {
```

Comment:
This summary is fairly specific to this gossip implementation. We could move it to the API and make the dump typed. I decided to do that for fetch because I think that's probably quite stable but this doesn't.

As it is, this is just returned as JSON.

Reply:
Why does it matter if its not stable? Holochain can still dump as json if we aren't ready to type its admin api dump responses.